corehq/apps/ota/views.py

```diff
@@ -264,22 +265,9 @@ def get_restore_response(domain, couch_user, app_id=None, since=None, version='1
 
     app = get_app_cached(domain, app_id) if app_id else None
 
-    #For ICDS only allow users who have access the specific app
-    if (
-        settings.SERVER_ENVIRONMENT in settings.ICDS_ENVS
-        and toggles.ROLE_WEBAPPS_PERMISSIONS.enabled(domain)
-    ):
-        has_access_to_app = False
-        try:
-            role = couch_user.get_role(domain)
-            if role is None:
-                return HttpResponse('User does not have any role assigned to it', status=406), None
-            has_access_to_app = role.permissions.view_web_app(app)
-            if not has_access_to_app:
-                return HttpResponse('User is not allowed on this app', status=406), None
-        except DomainMembershipError:
-            return HttpResponse('User does not belong to this domain', status=406), None
-
+    error_msg, status_code = check_authorization_errors(domain, couch_user, app)
```

Comment:
nit: I think this is not strictly "authorization" but "access" so renaming this to something like "check_app_access" would be more explicit about what its doing.
Plus I think the method can return an HttpResponse itself and then you can just update this to 
```
response = check_app_acccess(domain, couch_user, app)
if response:
    return response
```

Reply:
I was stumped on naming on this and then went with `check_authorization_errors ` 😆  
And I was struggling to choose between this approach and the one I implemented 😅 

Both seemed equally good to me, now you have also suggested returning `HttpResponse`, will do that.